server/storage/keyspace_test.go

```diff
@@ -0,0 +1,137 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"testing"
+	"time"
+
+	"github.com/pingcap/kvproto/pkg/keyspacepb"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeyspace(t *testing.T) {
+	re := require.New(t)
+	storage := NewStorageWithMemoryBackend()
+
+	keyspaces := testKeyspaces()
+	for _, keyspace := range keyspaces {
+		re.NoError(storage.SaveKeyspace(keyspace))
+	}
+
+	for _, keyspace := range keyspaces {
+		spaceID := keyspace.GetId()
+		loadedKeyspace := &keyspacepb.KeyspaceMeta{}
+		// Test load keyspace.
+		success, err := storage.LoadKeyspace(spaceID, loadedKeyspace)
+		re.True(success)
+		re.NoError(err)
+		re.Equal(keyspace, loadedKeyspace)
+		// Test remove keyspace.
+		re.NoError(storage.RemoveKeyspace(spaceID))
+		success, err = storage.LoadKeyspace(spaceID, loadedKeyspace)
+		// Loading a non-existing keyspace should be unsuccessful.
+		re.False(success)
+		// Loading a non-existing keyspace should not return error.
+		re.NoError(err)
+	}
+}
+
+func TestLoadRangeKeyspaces(t *testing.T) {
+	re := require.New(t)
+	storage := NewStorageWithMemoryBackend()
+
+	keyspaces := testKeyspaces()
+	for _, keyspace := range keyspaces {
+		re.NoError(storage.SaveKeyspace(keyspace))
+	}
+
+	// load all keyspaces.
+	loadedKeyspaces, err := storage.LoadRangeKeyspace(keyspaces[0].GetId(), 0)
+	re.NoError(err)
+	re.ElementsMatch(keyspaces, loadedKeyspaces)
+
+	// load keyspaces that with id no less than second test keyspace.
+	loadedKeyspaces2, err := storage.LoadRangeKeyspace(keyspaces[1].GetId(), 0)
+	re.NoError(err)
+	re.ElementsMatch(keyspaces[1:], loadedKeyspaces2)
+
+	// load keyspace with the smallest id.
+	loadedKeyspace3, err := storage.LoadRangeKeyspace(1, 1)
+	re.NoError(err)
+	re.ElementsMatch(keyspaces[:1], loadedKeyspace3)
+}
+
+func TestSaveLoadKeyspaceID(t *testing.T) {
+	re := require.New(t)
+	storage := NewStorageWithMemoryBackend()
+
+	ids := []uint32{100, 200, 300}
+	names := []string{"keyspace1", "keyspace2", "keyspace3"}
+	for i := range ids {
+		re.NoError(storage.SaveKeyspaceID(ids[i], names[i]))
+	}
+
+	for i := range names {
+		success, id, err := storage.LoadKeyspaceID(names[i])
+		re.NoError(err)
+		re.True(success)
+		re.Equal(ids[i], id)
+	}
+	// loading non-existing id should return false, 0, nil
+	success, id, err := storage.LoadKeyspaceID("non-existing")
+	re.NoError(err)
+	re.False(success)
+	re.Equal(uint32(0), id)
+}
+
+func testKeyspaces() []*keyspacepb.KeyspaceMeta {
```

Comment:
The function name needs to be able to simply describe the function logic.

Reply:
For example: https://github.com/tikv/pd/blob/34a4cce2cb07f8aa4283e28ba40a667e1321a1fb/server/api/region_label_test.go#L109